src/main/java/minijava/ir/utils/GraphUtils.java

```diff
@@ -61,17 +61,19 @@
   }
 
   public static boolean areConnected(Node source, Node target) {
-    final boolean[] connected = {false};
     Consumer<Node> visitor =
         n -> {
           if (n.equals(target)) {
-            connected[0] = true;
+            throw new EarlyExitException();
           }
         };
 
-    walkFromNodeDepthFirst(source, visitor, n -> {});
-
-    return connected[0];
+    try {
+      walkFromNodeDepthFirst(source, visitor, n -> {});
+      return false;
+    } catch (EarlyExitException e) {
+      return true;
+    }
   }
```

Comment:
Yeah, I know. I really hesitated with this and no, I haven't measured. I should revert it.